### PR TITLE
Fix complex rules for imag, real and abs

### PR DESCRIPTION
**Pull request by Pangoraw**
Depends on https://github.com/EnzymeAD/Enzyme/pull/2183.

### CODE REVIEW

**Comment by wsmoses** [comment]
okay now that the Enzyme PR is merged, you need to update the hash, otherwise lgtm

**Comment by Pangoraw** [comment]
it seems that bumping the Enzyme commit broke some CI builds, we need an xla bump in this case to update the llvm version as well ?

**Comment by wsmoses** [comment]
Our main might be slightly out of date tho so we may need to do some minor fix ups to enzyme every so often

File: src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp [hunk 1]
```diff
@@ -6172,6 +6173,26 @@ struct ImagOpCanon final : OpRewritePattern<mlir::stablehlo::ImagOp> {
   }
 };
 
+// (conj (complex a, (neg b))) -> (complex a b)
+struct ConjComplexNegate final : OpRewritePattern<mlir::chlo::ConjOp> {
```
Comment (wsmoses) [inline src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp:6177]: can this be added to the same default python hlo_opts string

**Comment by wsmoses** [comment]
bump was fixed here: https://github.com/EnzymeAD/Enzyme-JAX/pull/187 so you should rebase then be good ot go

**Review by wsmoses — APPROVED** [review]

**Pull request merged**